### PR TITLE
Sara/experiment/markprompt

### PR DESCRIPTION
**Pull request by s-santillan**
# Thanks for improving Semgrep Docs 😀

### Please ensure

- [ ] A subject matter expert (SME) reviews the content
- [ ] A technical writer reviews the content or PR

### Testing and reviewing this PR
(You must have an existing docusaurus installation in your machine.)

1. Pull the PR to your local development machine. 
2. From the root of your semgrep-docs repo, perform the following:
    1. Enter `yarn install`.
    2. Enter `yarn build`.
    3. Enter `git checkout docs/.`
3. Copy the development key from markprompt into docusaurus.config.js under `projectKey`. Do not commit the development key.
4. Run `yarn dev` to test Markprompt.

### CODE REVIEW

**Comment by pabloest** [comment]
@s-santillan I get a failure at the build stage 2ii:
```
ReferenceError: document is not defined
[ERROR] Unable to build website for locale en.
[ERROR] Error: Failed to compile with errors.
    at /Users/pablo/sites/semgrep-docs/node_modules/@docusaurus/core/lib/webpack/utils.js:180:24
    at /Users/pablo/sites/semgrep-docs/node_modules/webpack/lib/MultiCompiler.js:554:14
    at processQueueWorker (/Users/pablo/sites/semgrep-docs/node_modules/webpack/lib/MultiCompiler.js:491:6)
    at processTicksAndRejections (node:internal/process/task_queues:78:11)
```

What versions are you using for the following?
```
[INFO] Docusaurus version: 2.4.1
Node version: v16.14.2
```

**Comment by s-santillan** [comment]
@pabloest I tested this on a Mac, could you please try skipping 2ii. So please start with **2iii** and continue on from there. It should work.

File: docusaurus.config.js [hunk 1]
```diff
@@ -19,6 +19,9 @@ module.exports = {
         hideable: true,
       },
     },
+    markprompt: {
+      projectKey: 'jbhF5LligltdKaJucMjDcWcRodaVpzqE',
```
Comment (pabloest) [inline docusaurus.config.js:23]: @s-santillan intentional?

Comment (s-santillan) [reply docusaurus.config.js:23]: Yes. It's not the bearer token or the dev token (both of which should be either secrets or not committed at all). I apologize for the confusion.

**Comment by s-santillan** [comment]
Some tidbits about this PR:
1. We switch from using JS to TS (see [Docusaurus typescript support](https://docusaurus.io/docs/typescript-support)). This shouldn't affect anything in the docs themselves. The Markprompt component was only available as TS and I figured better to upgrade as TS is a superset of JS.
2. On Node v16.14.2 and Docusaurus 2.4.1 (same environment as me and Pablo) the build still consistently fails for *nix systems, I am looking into it now.

File: docusaurus.config.js (hunk 1 above)
Comment (pabloest) [reply docusaurus.config.js:23]: Thank you!

**Review by pabloest — APPROVED** [review]

**Pull request merged**